File: python/splitting.py
import pandas as pd

# Import dataset
df = pd.read_csv('data.csv')
X = df.iloc[:, :-1].values
y = df.iloc[:, -1].values

# Split into training and test set
# test_size specifies proportion of the test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)

# Apply 10 folds cross-validation
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import cross_val_score
model = LinearRegression()
scores = cross_val_score(model, X_train, y_train, cv=10)

# Report score
print("Accuracy %0.2f (+/- %0.2f)" % (scores.mean(), scores.std() * 2))

File: r/splitting.r
# Import dataset
df = read.csv('data.csv')

# caTools contains several basic utility functions including: moving 
# (rolling, running) window statistic functions, read/write for GIF 
# and ENVI binary files, fast calculation of AUC, LogitBoost classifier, 
# base64 encoder/decoder, round-off-error-free sum and cumsum, etc.
install.packages('caTools')
library(caTools)

# Create Boolean vector with random split
# SplitRatio specifies proportion of training set
split = sample.split(df$y, SplitRatio=0.8)

# Split into training and test set
train = subset(df, split == TRUE)
test = subset(df, split == FALSE)

# caret: Classification and Regression Training provides misc functions for 
# training and plotting classification and regression models.
install.packages('caret')
library(caret)

# Apply 10 folds cross-validation
folds = 10
cvIndex = createFolds(train$y, folds, returnTrain = TRUE)
tc = trainControl(index=cvIndex, method='cv', number=folds)